core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthChecker.java

```diff
@@ -16,14 +16,52 @@
 
 package com.linecorp.armeria.server.healthcheck;
 
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.annotation.UnstableApi;
 import com.linecorp.armeria.server.Server;
 
+import io.netty.util.concurrent.EventExecutor;
+
 /**
  * Determines whether the {@link Server} is healthy. All registered {@link HealthChecker}s must return
  * {@code true} for the {@link Server} to be considered healthy.
  */
 @FunctionalInterface
 public interface HealthChecker {
+
+    /**
+     * Create a {@link HealthChecker} which executing supplied health checker {@link Supplier} on a random
+     * {@link EventExecutor} from {@link CommonPools#workerGroup()} after constructing and subsequently with the
+     * given delay between the completion of supplied {@link CompletionStage} and the commencement of the next.
+     *
+     * @see #of(Supplier, Duration, EventExecutor)
+     */
+    @UnstableApi
+    static HealthChecker of(Supplier<? extends CompletionStage<HealthCheckStatus>> healthChecker,
+                            Duration fallbackTtl) {
+        return new ScheduledHealthChecker(healthChecker, fallbackTtl, CommonPools.workerGroup().next());
```

Comment:
- `requireNonNull` for all parameters?
- Check whether `fallbackTtl` is a positive duration.